pkg/fission-cli/cmd/spec/validate.go

```diff
@@ -72,6 +78,192 @@ func (opts *ValidateSubCommand) run(input cli.Input) error {
+	}
+	for _, p := range pkglist {
+		for _, pkg := range fr.Packages {
+			// if the enviornemt name is same check if they are present in the same namespace
+			if p.Metadata.Name == pkg.Metadata.Name {
+				if p.Metadata.Namespace == pkg.Metadata.Namespace {
+					result = multierror.Append(result, fmt.Errorf("\nPakcage %v/%v already exist", p.Metadata.Name, pkg.Metadata.Namespace))
+
+				}
+			}
+		}
+	}
+
+	httptriggerlist, err := getHTTPTriggers(opts)
+	if err != nil {
+		return errors.Errorf("Unable to get HTTPTrigger %v", err.Error())
+	}
+	for _, h := range httptriggerlist {
+		for _, htt := range fr.HttpTriggers {
+			// if the enviornemt name is same check if they are present in the same namespace
+			if h.Metadata.Name == htt.Metadata.Name {
+				if h.Metadata.Namespace == htt.Metadata.Namespace {
+					result = multierror.Append(result, fmt.Errorf("\n HttpTrigger %v/%v already exist", htt.Metadata.Name, htt.Metadata.Namespace))
+
+				}
+			}
+		}
+	}
+	mqtriggerlist, err := getMessageQueueTriggers(opts, "")
+	if err != nil {
+		return errors.Errorf("Unable to get Message Queue Trigger %v", err.Error())
+	}
+	for _, m := range mqtriggerlist {
+		for _, mqt := range fr.MessageQueueTriggers {
+			// if the messagequeue trigger name is same check if they are present in the same namespace
+			if m.Metadata.Name == mqt.Metadata.Name {
+				if m.Metadata.Namespace == mqt.Metadata.Namespace {
+					result = multierror.Append(result, fmt.Errorf("\n MessageQueueTriggers %v/%v already exist", m.Metadata.Name, mqt.Metadata.Namespace))
+
+				}
+			}
+		}
+	}
+
+	timetriggerlist, err := getTimeTriggers(opts)
+	if err != nil {
+		return errors.Errorf("Unable to get Time Trigger %v", err.Error())
+	}
+	for _, t := range timetriggerlist {
+		for _, tt := range fr.MessageQueueTriggers {
+			// if the enviornemt name is same check if they are present in the same namespace
+			if t.Metadata.Name == tt.Metadata.Name {
+				if t.Metadata.Namespace == tt.Metadata.Namespace {
+					result = multierror.Append(result, fmt.Errorf("\n TimeTrigger %v/%v already exist", tt.Metadata.Name, tt.Metadata.Namespace))
+
+				}
+			}
+		}
+	}
+
+	kubewatchtriggerlist, err := getKubeWatchTriggers(opts)
+	if err != nil {
+		return errors.Errorf("Unable to get Kubernetes Watch Trigger %v", err.Error())
+	}
+	for _, k := range kubewatchtriggerlist {
+		for _, kwt := range fr.KubernetesWatchTriggers {
+			// if the enviornemt name is same check if they are present in the same namespace
+			if k.Metadata.Name == kwt.Metadata.Name {
+				if k.Metadata.Namespace == kwt.Metadata.Namespace {
+					result = multierror.Append(result, fmt.Errorf("\n Kubernetes Watch Trigger %v/%v already exist", kwt.Metadata.Name, kwt.Metadata.Namespace))
+
+				}
+			}
+		}
+	}
+	return result.ErrorOrNil()
+}
+
+// getAllFunctions get lists of functions in all namespaces
+func getFunctions(opts *ValidateSubCommand) ([]fv1.Function, error) {
```

Comment:
Since we already have similar functions in `spec/list.go`, we should reuse them instead of adding duplicate functions.

https://github.com/fission/fission/blob/66d4669f364cc5509e8a895cd2b5680e91ba0f29/pkg/fission-cli/cmd/spec/list.go#L384-L454

We can modify the signature of the existing function to something like following
```
import "github.com/fission/fission/pkg/controller/client"

func getAllKubeWatchTriggers(client client.Interface) ([]fv1.KubernetesWatchTrigger, error) {
	ws, err := client.V1().KubeWatcher().List("")
        ...
```

**Please always consider reusing the existing code first unless it's impossible to reuse them.**

Reply:
Yeah sure.